charms/istio-gateway/tests/test_charm.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+from charm import Operator
+from ops.model import BlockedStatus, WaitingStatus
+from ops.testing import Harness
+
+
+@pytest.fixture
+def harness():
+    return Harness(Operator)
+
+
+def test_not_leader(harness):
+    harness.begin()
+    assert harness.charm.model.unit.status == WaitingStatus('Waiting for leadership')
+
+
+def test_basic(harness):
+    harness.set_leader(True)
+    harness.begin_with_initial_hooks()
+    assert harness.charm.model.unit.status == BlockedStatus('Config item `kind` must be set')
```

Comment:
An additional test to make sure `kubectl` is called correctly would be nice here.